crates/router/src/connector/stripe/transformers.rs

```diff
@@ -1207,25 +1205,16 @@ pub struct StripeRedirectResponse {
     pub source_type: Option<Secret<String>>,
 }
 
-#[derive(Debug, Serialize, Clone, Copy)]
+#[derive(Debug, Serialize)]
 pub struct CancelRequest {
-    cancellation_reason: Option<CancellationReason>,
+    cancellation_reason: Option<String>,
```

Comment:
The enum `CancellationReason` is not deleted.

Reply:
@Abhicodes-crypto 

RequestEncodingFailedWithReason should be handled only as InternalServerError as this is a failure happened while building connector request at our side.

However, we can solve this problems in two ways,
1. Having validation at our side (using enums).   Incase validation fails we can have a new error type which will propagate proper message instead of throwing `InternalServerError`. But till now we don't have any unified logic to handle connector level field validation. And we have to maintain a proper documentation for the possible validation errors.Incase validation fails we can have a new error type which will propagate proper message instead of throwing `InternalServerError`. But till now we don't have any unified logic to handle connector level field validation. And we have to maintain a proper documentation for the possible validation errors.

2. Relying on connector's validation. Exact error message from connector will be shown as ConnectorError in the response instead of us throwing error on behalf of connector.

I discussed both the solution with @jarnura, Considering the complexity involves in first he suggested to go with 2 solution.

Reply:
Cool. 

I guess you can still delete `enum CancellationReason` if it is not being used anywhere in `stripe/transformers.rs`.

Also I guess an appropriate error message could be given here. 
https://github.com/juspay/hyperswitch/blob/cf902f19e5e08c6ee538ab4be7411f50a4079b55/crates/router/src/connector/stripe.rs#L558

Reply:
Yeah you are right. In stripe.rs it should be RequestEncodingFailed error. I will change there too